crates/evm/src/system_calls.rs

```diff
@@ -264,3 +265,138 @@ where
 
     Ok(withdrawal_requests)
 }
+
+/// Apply the [EIP-7251](https://eips.ethereum.org/EIPS/eip-7251) post block contract call.
+///
+/// This constructs a new [Evm] with the given DB, and environment
+/// ([`CfgEnvWithHandlerCfg`] and [`BlockEnv`]) to execute the post block contract call.
+///
+/// This uses [`apply_consolidation_requests_contract_call`] to ultimately calculate the
+/// [requests](Request).
+pub fn post_block_consolidation_requests_contract_call<EvmConfig, DB>(
```

Comment:
we have essentially the same functionality copy pasted 3 times now, maybe perhaps time to abstract it